src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java

```diff
@@ -338,12 +325,133 @@ public void storeContext(Context context, HttpServletRequest request) {
+			return isAdmin;
+		}
+
+		JSONArray subAccountAdminsJson = new JSONArray(subAccountAdminsResponse);
+		if (subAccountAdminsJson.length() > 0) {
+			//user is a some kind of admin in the course
+			M_log.info(String.format("User %s is an admin",userCanvasId));
+			isAdmin = true;
+			return isAdmin;
+		}
+		//find out if the user is a subaccount admin one level up in the hierarchy
+		int parentAccount = getParentAccount(courseAccountId);
+		// 0 means cannot get the desired response or trying to get parent of root account which is not applicable
+		if (parentAccount == 0) {
+			return isAdmin;
+		}
+		return determineIfUserIsAdminInHierarchy(userCanvasId, parentAccount);
+	}
+
+	public int getParentAccount(int accountId){
+		M_log.debug("getParentAccount() call");
+		// https://umich.test.instructure.com:443/api/v1/accounts/78
+		String url = canvasURL + "/api/v1/accounts/" + accountId;
+		M_log.info("API call "+url);
+		HttpResponse response = processApiCall(new HttpGet(url));
+		if (response == null) {
+			M_log.error(String.format("getting parent for account %s for SubAccountAdmin check failed due " +
+					"to null response", accountId));
+			return 0;
+		}
+		int statusCode = response.getStatusLine().getStatusCode();
+		if (statusCode != HttpStatus.SC_OK) {
+			M_log.error(String.format("getting parent for account %s for SubAccountAdmin check failed " +
+					"with status %s and " + "with error message", accountId, statusCode, response.toString()));
+			return 0;
+		}
+
+		String parentAccountResponse;
+		int parentAccountId = 0;
+		String errMsg = "As part of SubAccountAdmin check couldn't get " +
+				"the ParentId for account %s due to %s";
+		try {
+			parentAccountResponse = EntityUtils.toString(response.getEntity(), "UTF-8");
+			JSONObject parentAccountJson = new JSONObject(parentAccountResponse);
+			if (!parentAccountJson.isNull("parent_account_id")) {
+				parentAccountId = (int) parentAccountJson.get("parent_account_id");
+			}
+		} catch (IOException e) {
+			M_log.error(String.format(errMsg, accountId, e.getMessage()));
+			return 0;
+		} catch (Exception e) {
+			M_log.error(String.format(errMsg, accountId, e.getMessage()));
+			return 0;
+		}
+		return parentAccountId;
+
+	}
+
+
+	private int getCourseAccountId(String courseId){
+		M_log.debug("getCourseAccountId() call");
+		int course_account_id = 0;
+		String url = canvasURL + "/api/v1/courses/" + courseId;
+		M_log.info("API call "+url);
+		HttpResponse response = processApiCall(new HttpGet(url));
+		if (response == null) {
+			return 0;
+		}
+		int statusCode = response.getStatusLine().getStatusCode();
+		if (statusCode != HttpStatus.SC_OK) {
+			return course_account_id;
+		}
+		String courseData = null;
+		try {
+			courseData = EntityUtils.toString(response.getEntity(), "UTF-8");
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+		JSONObject courseJson = new JSONObject(courseData);
+		course_account_id = (int) courseJson.get("account_id");
```

Comment:
I have added a null check just for safety i don't think that value will be null.
`if (!courseJson.isNull("account_id")) {
			course_account_id = (int) courseJson.get("account_id");
		}`